src/System.Private.CoreLib/src/System/Globalization/FormatProvider.FormatAndParse.Unix.cs

```diff
@@ -0,0 +1,164 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime;
+using System.Globalization;
+using System.Diagnostics;
+
+namespace System.Globalization
+{
+    internal partial class FormatProvider
+    {
+        private partial class Number
+        {
+            // buffer size to hold digits (40), decimal point, number sign, exponent, exponent symbol 'e', exponent sign and null.
+            private const int  ECVT_MAX_BUFFER_SIZE = 50;  
+           
+            private static unsafe void DoubleToNumber(double value, int precision, ref NumberBuffer number)
+            {
+                Debug.Assert(precision > 0 && precision < 40);
+                
+                number.precision = precision;
+                if (DoubleHelper.Exponent(value) == 0x7ff)
+                {
+                    number.scale = DoubleHelper.Mantissa(value) != 0 ? SCALE_NAN : SCALE_INF;
+                    number.sign = DoubleHelper.Sign(value);
+                    number.digits[0] = '\0';
+                    return;
+                }
+                
+                byte* tempBuffer = stackalloc byte[ECVT_MAX_BUFFER_SIZE];
+                char* dst = number.digits;
+
+                number.scale = 0;
+                number.sign = false;
+                *dst = '\0';             
+
+                if (value < 0.0)
+                {
+                    number.sign = true;
+                }
+                
+                if (value == 0.0)
+                {
+                    for (int j=0; j<precision; j++)
+                    {
+                        dst[j] = '0';
+                    }
+                    dst[precision] = '\0';
+                    return;
+                }
+
+                //
+                // Get the number formatted as a string in the form x.xxxxxxexxxx
+                //
+
+                // "%.40e"
+                uint *format = stackalloc uint[2];
```

Comment:
This does not work on bigendian machines. We should allocate array of bytes, and assign it one byte at a time - I think it will be also more readable that way.